mac debug build broken

introduced because of prebuilt v8: 89096c92104ab89b1fd8ba273cbbf441dcfe5a84

fixed with this small change: https://github.com/denoland/deno/pull/1218/commits/e5f204fe49ef492ab97bdc4dff43be731dbbde25

error looks like:
```
Undefined symbols for architecture x86_64:
  "_GFp_BN_mod_exp_mont_consttime", referenced from:
      ring::rsa::bigint::elem_exp_consttime::h2efacbbd9e4e21e0 in libring.rlib(ring.ring0.rcgu.o)
      ring::rsa::bigint::elem_exp_consttime::h3426a412925f976d in libring.rlib(ring.ring0.rcgu.o)
  "_GFp_ChaCha20_ctr32", referenced from:
      ring::chacha::chacha20_xor_inner::hf30e5b063ebef8b4 in libring.rlib(ring.ring3.rcgu.o)
  "_GFp_aes_gcm_init", referenced from:
      ring::aead::aes_gcm::aes_gcm_init::ha96301199aec9555 in libring.rlib(ring.ring1.rcgu.o)
  "_GFp_aes_gcm_open", referenced from:
      ring::aead::aes_gcm::aes_gcm_open::h14dd843bc8019d2c in libring.rlib(ring.ring1.rcgu.o)
  "_GFp_aes_gcm_seal", referenced from:
      ring::aead::aes_gcm::aes_gcm_seal::he6127808aee18c16 in libring.rlib(ring.ring1.rcgu.o)
  "_GFp_bn_from_montgomery_in_place", referenced from:
      ring::rsa::bigint::elem_reduced::h335920a66181b2ca in libring.rlib(ring.ring0.rcgu.o)
      ring::rsa::bigint::elem_reduced::hfd1cb7935a230833 in libring.rlib(ring.ring0.rcgu.o)
  "_GFp_bn_mul_mont", referenced from:
      _$LT$ring..rsa..bigint..Elem$LT$M$C$$u20$E$GT$$GT$::decode_once::h729f2dd7e9b20fa6 in libring.rlib(ring.ring0.rcgu.o)
      _$LT$ring..rsa..bigint..Elem$LT$M$C$$u20$E$GT$$GT$::decode_once::hd38f8b3009519726 in libring.rlib(ring.ring0.rcgu.o)
      _$LT$ring..rsa..bigint..Elem$LT$M$C$$u20$E$GT$$GT$::decode_once::hde9891061745b4c4 in libring.rlib(ring.ring0.rcgu.o)
      ring::rsa::bigint::elem_mul::h0cca3d12252441fa in libring.rlib(ring.ring0.rcgu.o)
      ring::rsa::bigint::elem_mul::h0fbdbc0b7b58b0bd in libring.rlib(ring.ring0.rcgu.o)
      ring::rsa::bigint::elem_mul::h16fb48cdfef13061 in libring.rlib(ring.ring0.rcgu.o)
      ring::rsa::bigint::elem_mul::h3e9c6c465e5d019c in libring.rlib(ring.ring0.rcgu.o)
      ...
  "_GFp_bn_mul_mont_check_num_limbs", referenced from:
      _$LT$ring..rsa..bigint..Modulus$LT$M$GT$$GT$::from_boxed_limbs::h6022c58e668aa065 in libring.rlib(ring.ring0.rcgu.o)
      _$LT$ring..rsa..bigint..Modulus$LT$M$GT$$GT$::from_boxed_limbs::h67711c5b43ee16ff in libring.rlib(ring.ring0.rcgu.o)
      _$LT$ring..rsa..bigint..Modulus$LT$M$GT$$GT$::from_boxed_limbs::ha4b167d513256fc8 in libring.rlib(ring.ring0.rcgu.o)
      _$LT$ring..rsa..bigint..Modulus$LT$M$GT$$GT$::from_boxed_limbs::hadb2fe63a695d454 in libring.rlib(ring.ring0.rcgu.o)
  "_GFp_bn_neg_inv_mod_r_u64", referenced from:
      _$LT$ring..rsa..bigint..Modulus$LT$M$GT$$GT$::from_boxed_limbs::h6022c58e668aa065 in libring.rlib(ring.ring0.rcgu.o)
      _$LT$ring..rsa..bigint..Modulus$LT$M$GT$$GT$::from_boxed_limbs::h67711c5b43ee16ff in libring.rlib(ring.ring0.rcgu.o)
      _$LT$ring..rsa..bigint..Modulus$LT$M$GT$$GT$::from_boxed_limbs::ha4b167d513256fc8 in libring.rlib(ring.ring0.rcgu.o)
      _$LT$ring..rsa..bigint..Modulus$LT$M$GT$$GT$::from_boxed_limbs::hadb2fe63a695d454 in libring.rlib(ring.ring0.rcgu.o)
  "_GFp_cpuid_setup", referenced from:
      ring::init::init_once::_$u7b$$u7b$closure$u7d$$u7d$::hedc6f60cf5281be0 in libring.rlib(ring.ring1.rcgu.o)
  "_GFp_memcmp", referenced from:
      ring::constant_time::verify_slices_are_equal::hb8d4db7f7784f07e in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_nistz256_add", referenced from:
      ring::ec::suite_b::ops::p256::COMMON_OPS::h84656d1efcc49c76 in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_nistz256_mul_mont", referenced from:
      ring::ec::suite_b::ops::p256::COMMON_OPS::h84656d1efcc49c76 in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_nistz256_point_add", referenced from:
      ring::ec::suite_b::ops::p256::COMMON_OPS::h84656d1efcc49c76 in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_nistz256_point_mul", referenced from:
      ring::ec::suite_b::ops::p256::PRIVATE_KEY_OPS::h8f822d1aaf4b4c7e in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_nistz256_point_mul_base", referenced from:
      ring::ec::suite_b::ops::p256::p256_point_mul_base_impl::hbca360dcfbe9df03 in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_nistz256_sqr_mont", referenced from:
      ring::ec::suite_b::ops::p256::COMMON_OPS::h84656d1efcc49c76 in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_nistz384_point_add", referenced from:
      ring::ec::suite_b::ops::p384::COMMON_OPS::he95591db372a3360 in libring.rlib(ring.ring7.rcgu.o)
  "_GFp_nistz384_point_mul", referenced from:
      ring::ec::suite_b::ops::p384::PRIVATE_KEY_OPS::h8a0fd939cd0f657d in libring.rlib(ring.ring7.rcgu.o)
  "_GFp_p256_scalar_mul_mont", referenced from:
      ring::ec::suite_b::ops::p256::p256_scalar_inv_to_mont::mul::ha8f8ba11e093df73 in libring.rlib(ring.ring6.rcgu.o)
      ring::ec::suite_b::ops::p256::p256_scalar_inv_to_mont::sqr_mul_acc::h935b7549d66d52d9 in libring.rlib(ring.ring6.rcgu.o)
      ring::ec::suite_b::ops::p256::p256_scalar_inv_to_mont::to_mont::he1d4cdb03e5fd9fb in libring.rlib(ring.ring6.rcgu.o)
      ring::ec::suite_b::ops::p256::SCALAR_OPS::hd8d3f6cb0cb75969 in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_p256_scalar_sqr_mont", referenced from:
      ring::ec::suite_b::ops::p256::p256_scalar_inv_to_mont::sqr::h6047b3283ade120a in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_p256_scalar_sqr_rep_mont", referenced from:
      ring::ec::suite_b::ops::p256::p256_scalar_inv_to_mont::sqr_mul::hd7c34afbd45c4386 in libring.rlib(ring.ring6.rcgu.o)
      ring::ec::suite_b::ops::p256::p256_scalar_inv_to_mont::sqr_mul_acc::h935b7549d66d52d9 in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_p384_elem_add", referenced from:
      ring::ec::suite_b::ops::p384::COMMON_OPS::he95591db372a3360 in libring.rlib(ring.ring7.rcgu.o)
  "_GFp_p384_elem_mul_mont", referenced from:
      ring::ec::suite_b::ops::p384::GFp_p384_elem_sqr_mont::h7e0f2d5dc9810eef in libring.rlib(ring.ring7.rcgu.o)
      ring::ec::suite_b::ops::p384::COMMON_OPS::he95591db372a3360 in libring.rlib(ring.ring7.rcgu.o)
  "_GFp_p384_scalar_mul_mont", referenced from:
      ring::ec::suite_b::ops::p384::p384_scalar_inv_to_mont::mul::h5ede5a094aeced52 in libring.rlib(ring.ring7.rcgu.o)
      ring::ec::suite_b::ops::p384::p384_scalar_inv_to_mont::sqr::heb31b88e5c7dfb1d in libring.rlib(ring.ring7.rcgu.o)
      ring::ec::suite_b::ops::p384::p384_scalar_inv_to_mont::sqr_mut::hd7f4af2dc7688f8a in libring.rlib(ring.ring7.rcgu.o)
      ring::ec::suite_b::ops::p384::p384_scalar_inv_to_mont::sqr_mul_acc::h4bdc99023c95b324 in libring.rlib(ring.ring7.rcgu.o)
      ring::ec::suite_b::ops::p384::p384_scalar_inv_to_mont::to_mont::h51d783aa7981ee68 in libring.rlib(ring.ring7.rcgu.o)
      ring::ec::suite_b::ops::p384::SCALAR_OPS::h6a47e39b7f102597 in libring.rlib(ring.ring7.rcgu.o)
  "_GFp_poly1305_blocks", referenced from:
      ring::poly1305::SigningContext::from_key::hc874d662904d74ef in libring.rlib(ring.ring3.rcgu.o)
  "_GFp_poly1305_emit", referenced from:
      ring::poly1305::SigningContext::from_key::hc874d662904d74ef in libring.rlib(ring.ring3.rcgu.o)
  "_GFp_poly1305_init_asm", referenced from:
      ring::poly1305::init::hf7ac8b932f675edc in libring.rlib(ring.ring3.rcgu.o)
  "_GFp_sha256_block_data_order", referenced from:
      ring::digest::SHA256::hdb690df8c8bd7b61 in libring.rlib(ring.ring12.rcgu.o)
  "_GFp_sha512_block_data_order", referenced from:
      ring::digest::SHA384::h32931267b7502c85 in libring.rlib(ring.ring12.rcgu.o)
      ring::digest::SHA512::h159a93b7323ece03 in libring.rlib(ring.ring12.rcgu.o)
      ring::digest::SHA512_256::h027f5e82497d2592 in libring.rlib(ring.ring12.rcgu.o)
  "_GFp_x25519_fe_invert", referenced from:
      ring::ec::curve25519::ops::encode_point::h591a8bd7ffd8e2cc in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_x25519_fe_isnegative", referenced from:
      ring::ec::curve25519::ops::encode_point::h591a8bd7ffd8e2cc in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_x25519_fe_mul_ttt", referenced from:
      ring::ec::curve25519::ops::encode_point::h591a8bd7ffd8e2cc in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_x25519_fe_neg", referenced from:
      _$LT$ring..ec..curve25519..ops..Elem$LT$ring..ec..curve25519..ops..T$GT$$GT$::negate::he4d19bf3740d26f4 in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_x25519_fe_tobytes", referenced from:
      ring::ec::curve25519::ops::encode_point::h591a8bd7ffd8e2cc in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_x25519_ge_double_scalarmult_vartime", referenced from:
      _$LT$ring..ec..curve25519..ed25519..EdDSAParameters$u20$as$u20$ring..signature..VerificationAlgorithm$GT$::verify::h3b5af8e6e66a0686 in libring.rlib(ring.ring13.rcgu.o)
  "_GFp_x25519_ge_frombytes_vartime", referenced from:
      ring::ec::curve25519::ops::ExtPoint::from_encoded_point_vartime::h549e6ef94027c7c1 in libring.rlib(ring.ring6.rcgu.o)
  "_GFp_x25519_ge_scalarmult_base", referenced from:
      ring::ec::curve25519::ed25519::KeyPair::from_seed_::h1c78ca915f081e82 in libring.rlib(ring.ring13.rcgu.o)
      ring::ec::curve25519::ed25519::KeyPair::sign::h5d0703a7f09e68da in libring.rlib(ring.ring13.rcgu.o)
  "_GFp_x25519_public_from_private", referenced from:
      ring::ec::curve25519::x25519::x25519_public_from_private::h017a53f65b795ada in libring.rlib(ring.ring1.rcgu.o)
  "_GFp_x25519_sc_mask", referenced from:
      ring::ec::curve25519::ed25519::KeyPair::from_seed_::h1c78ca915f081e82 in libring.rlib(ring.ring13.rcgu.o)
  "_GFp_x25519_sc_muladd", referenced from:
      ring::ec::curve25519::ed25519::KeyPair::sign::h5d0703a7f09e68da in libring.rlib(ring.ring13.rcgu.o)
  "_GFp_x25519_sc_reduce", referenced from:
      ring::ec::curve25519::ed25519::digest_scalar::h2a8ce959dce90d21 in libring.rlib(ring.ring13.rcgu.o)
  "_GFp_x25519_scalar_mult", referenced from:
      ring::ec::curve25519::x25519::x25519_ecdh::hfe37aa4973f8535a in libring.rlib(ring.ring1.rcgu.o)
  "_LIMBS_add_mod", referenced from:
      ring::ec::suite_b::ops::scalar_sum::h1f0669698aefbdda in libring.rlib(ring.ring9.rcgu.o)
      ring::rsa::bigint::elem_add::h4074c2175df6ec70 in libring.rlib(ring.ring0.rcgu.o)
      ring::rsa::bigint::elem_add::hed534a6ad089d5b4 in libring.rlib(ring.ring0.rcgu.o)
  "_LIMBS_are_even", referenced from:
      ring::limb::limbs_are_even_constant_time::h565420e6a4b9b7f5 in libring.rlib(ring.ring13.rcgu.o)
  "_LIMBS_are_zero", referenced from:
      ring::limb::limbs_are_zero_constant_time::hf15626161850aa87 in libring.rlib(ring.ring13.rcgu.o)
  "_LIMBS_equal_limb", referenced from:
      ring::limb::limbs_equal_limb_constant_time::h47832576b2792d15 in libring.rlib(ring.ring13.rcgu.o)
  "_LIMBS_less_than", referenced from:
      ring::limb::limbs_less_than_limbs_consttime::hd0cbd73bf275c153 in libring.rlib(ring.ring13.rcgu.o)
  "_LIMBS_less_than_limb", referenced from:
      ring::limb::limbs_less_than_limb_constant_time::hb63585a939c1c791 in libring.rlib(ring.ring13.rcgu.o)
  "_LIMBS_reduce_once", referenced from:
      ring::limb::limbs_reduce_once_constant_time::h82b3059038ba84c0 in libring.rlib(ring.ring13.rcgu.o)
  "_LIMBS_shl_mod", referenced from:
      _$LT$ring..rsa..bigint..One$LT$M$C$$u20$ring..arithmetic..montgomery..RR$GT$$GT$::newRR::h24e48d41b3b4da86 in libring.rlib(ring.ring0.rcgu.o)
      _$LT$ring..rsa..bigint..One$LT$M$C$$u20$ring..arithmetic..montgomery..RR$GT$$GT$::newRR::h2fabef6590fd6171 in libring.rlib(ring.ring0.rcgu.o)
      _$LT$ring..rsa..bigint..One$LT$M$C$$u20$ring..arithmetic..montgomery..RR$GT$$GT$::newRR::h59eef47c3ed00450 in libring.rlib(ring.ring0.rcgu.o)
  "_LIMBS_sub_mod", referenced from:
      ring::rsa::bigint::elem_sub::hda91498f8ca22403 in libring.rlib(ring.ring0.rcgu.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Traceback (most recent call last):
  File "../../build/toolchain/mac/linker_driver.py", line 229, in <module>
    Main(sys.argv)
  File "../../build/toolchain/mac/linker_driver.py", line 79, in Main
    subprocess.check_call(compiler_driver_args)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 540, in check_call
```

--- a/BUILD.gn
+++ b/BUILD.gn
@@ -1,20 +1,20 @@
 # Copyright 2018 the Deno authors. All rights reserved. MIT license.
 import("//build/toolchain/cc_wrapper.gni")
-import("//third_party/v8/gni/v8.gni")
-import("//third_party/v8/snapshot_toolchain.gni")
 import("//build_extra/flatbuffers/flatbuffer.gni")
 import("//build_extra/flatbuffers/rust/rust_flatbuffer.gni")
-import("//build_extra/deno.gni")
 import("//build_extra/rust/rust.gni")
 import("//build_extra/toolchain/validate.gni")
+import("//third_party/v8/gni/v8.gni")
+import("//third_party/v8/snapshot_toolchain.gni")
+import("libdeno/deno.gni")
 
 group("default") {
   testonly = true
   deps = [
     ":deno",
     ":hyper_hello",
-    ":test_cc",
     ":test_rs",
+    "libdeno:test_cc",
   ]
 }
 
@@ -23,31 +23,6 @@ group("cargo_check_deps") {
   deps = [
     ":msg_rs",
   ]
-}
-
-config("deno_config") {
-  include_dirs = [ "third_party/v8" ]  # This allows us to v8/src/base/ libraries.
-  configs = [ "third_party/v8:external_config" ]
-  if (is_debug) {
-    defines = [ "DEBUG" ]
-  }
-
-  # Targets built with the `rust_executable()` template automatically pick up
-  # these dependencies, but those built with `executable()` need them when they
-  # have Rust inputs. Currently, there's only one such target, `test_cc`.
-  if (is_mac) {
-    libs = [ "resolv" ]
-  }
-  if (is_win) {
-    libs = [ "userenv.lib" ]
-  }
-
-  if (is_clang) {
-    cflags = [
-      "-fcolor-diagnostics",
-      "-fansi-escape-codes",
-    ]
-  }
 }
 
 main_extern = [
@@ -149,7 +124,7 @@ source_set("snapshot") {
   sources = [
     "src/snapshot.cc",
   ]
-  configs += [ ":deno_config" ]
+  configs += [ "libdeno:deno_config" ]
   inputs = [
     "$target_gen_dir/snapshot_deno.bin",
   ]
@@ -198,89 +173,14 @@ rust_test("test_rs") {
   ]
 }
 
-v8_executable("test_cc") {
-  testonly = true
-  sources = [
-    "libdeno/file_util_test.cc",
-    "libdeno/libdeno_test.cc",
-    "libdeno/test.cc",
-  ]
-  deps = [
-    ":create_snapshot_libdeno_test",
-    ":libdeno",
-    "//testing/gtest:gtest",
-  ]
-  data = [
-    "$target_gen_dir/snapshot_libdeno_test.bin",
-  ]
-  snapshot_path = rebase_path(data[0], root_build_dir)
-  defines = [ "SNAPSHOT_PATH=\"$snapshot_path\"" ]
-  configs = [ ":deno_config" ]
-}
-
-v8_static_library("v8") {
-  public_deps = [
-    "third_party/v8:v8",
-    "third_party/v8:v8_libbase",
-    "third_party/v8:v8_libplatform",
-    "third_party/v8:v8_libsampler",
-    "//build/win:default_exe_manifest",
-  ]
-  configs = [ ":deno_config" ]
-}
-
-# Only functionality needed for libdeno_test and snapshot_creator
-# In particular no flatbuffers, no assets, no rust, no msg handlers.
-# Because snapshots are slow, it's important that snapshot_creator's
-# dependencies are minimal.
-v8_static_library("libdeno") {
-  configs = [ ":deno_config" ]
-  sources = [
-    "libdeno/api.cc",
-    "libdeno/binding.cc",
-    "libdeno/deno.h",
-    "libdeno/file_util.cc",
-    "libdeno/file_util.h",
-    "libdeno/internal.h",
-  ]
-  if (!use_prebuilt_v8) {
-    public_deps = [
-      ":v8",
-    ]
-  } else {
-    # TODO(ry) It would be nice to have a standalone target for the prebuilt
-    # library that could simply be added to the deps here, but it wasn't
-    # obvious how to accomplish that in gn.
-    if (is_mac) {
-      libs = [ "//prebuilt/mac/libv8.a" ]
-    } else if (is_linux) {
-      libs = [ "//prebuilt/linux64/libv8.a" ]
-    } else if (is_win) {
-      libs = [ "//prebuilt/win/v8.lib" ]
-    } else {
-      assert(false, "We don't have prebuilt binaries for this platform yet.")
-    }
-  }
-}
-
 v8_static_library("deno_deps") {
   complete_static_lib = true
   public_deps = [
-    ":libdeno",
     ":msg_rs",
     ":snapshot",
+    "libdeno:libdeno",
   ]
-  configs = [ ":deno_config" ]
-}
-
-v8_executable("snapshot_creator") {
-  sources = [
-    "libdeno/snapshot_creator.cc",
-  ]
-  deps = [
-    ":libdeno",
-  ]
-  configs = [ ":deno_config" ]
+  configs = [ "libdeno:deno_config" ]
 }
 
 # Generates the core TypeScript type library for deno that will be
@@ -379,10 +279,4 @@ create_snapshot("deno") {
   deps = [
     ":bundle",
   ]
-}
-
-# Generates $target_gen_dir/snapshot_libdeno_test.bin
-create_snapshot("libdeno_test") {
-  testonly = true
-  js = "libdeno/libdeno_test.js"
 }

--- a/build_extra/rust/BUILD.gn
+++ b/build_extra/rust/BUILD.gn
@@ -726,7 +726,7 @@ rust_crate("dirs") {
 
 ring_root = "$registry_github/ring-0.13.2/"
 
-component("ring_primitives") {
+static_library("ring_primitives") {
   sources = [
     "$ring_root/crypto/constant_time_test.c",
     "$ring_root/crypto/cpu-aarch64-linux.c",

--- a/libdeno/BUILD.gn
+++ b/libdeno/BUILD.gn
@@ -1,0 +1,98 @@
+# Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import("./deno.gni")
+import("//third_party/v8/gni/v8.gni")
+
+config("deno_config") {
+  include_dirs = [ "//third_party/v8" ]  # This allows us to v8/src/base/ libraries.
+  configs = [ "//third_party/v8:external_config" ]
+  if (is_debug) {
+    defines = [ "DEBUG" ]
+  }
+  if (is_clang) {
+    cflags = [
+      "-fcolor-diagnostics",
+      "-fansi-escape-codes",
+    ]
+  }
+}
+
+v8_static_library("v8") {
+  public_deps = [
+    "//build/win:default_exe_manifest",
+    "//third_party/v8:v8",
+    "//third_party/v8:v8_libbase",
+    "//third_party/v8:v8_libplatform",
+    "//third_party/v8:v8_libsampler",
+  ]
+  configs = [ ":deno_config" ]
+}
+
+# Only functionality needed for libdeno_test and snapshot_creator
+# In particular no flatbuffers, no assets, no rust, no msg handlers.
+# Because snapshots are slow, it's important that snapshot_creator's
+# dependencies are minimal.
+v8_static_library("libdeno") {
+  configs = [ ":deno_config" ]
+  sources = [
+    "api.cc",
+    "binding.cc",
+    "deno.h",
+    "file_util.cc",
+    "file_util.h",
+    "internal.h",
+  ]
+  if (!use_prebuilt_v8) {
+    public_deps = [
+      ":v8",
+    ]
+  } else {
+    # TODO(ry) It would be nice to have a standalone target for the prebuilt
+    # library that could simply be added to the deps here, but it wasn't
+    # obvious how to accomplish that in gn.
+    if (is_mac) {
+      libs = [ "//prebuilt/mac/libv8.a" ]
+    } else if (is_linux) {
+      libs = [ "//prebuilt/linux64/libv8.a" ]
+    } else if (is_win) {
+      libs = [ "//prebuilt/win/v8.lib" ]
+    } else {
+      assert(false, "We don't have prebuilt binaries for this platform yet.")
+    }
+  }
+}
+
+v8_executable("snapshot_creator") {
+  sources = [
+    "snapshot_creator.cc",
+  ]
+  deps = [
+    ":libdeno",
+  ]
+  configs = [ ":deno_config" ]
+}
+
+v8_executable("test_cc") {
+  testonly = true
+  sources = [
+    "file_util_test.cc",
+    "libdeno_test.cc",
+    "test.cc",
+  ]
+  deps = [
+    ":create_snapshot_libdeno_test",
+    ":libdeno",
+    "//testing/gtest:gtest",
+  ]
+  data = [
+    "$target_gen_dir/snapshot_libdeno_test.bin",
+  ]
+  snapshot_path = rebase_path(data[0], root_build_dir)
+  defines = [ "SNAPSHOT_PATH=\"$snapshot_path\"" ]
+  configs = [ ":deno_config" ]
+}
+
+# Generates $target_gen_dir/snapshot_libdeno_test.bin
+create_snapshot("libdeno_test") {
+  testonly = true
+  js = "libdeno_test.js"
+}

--- a/libdeno/deno.gni
+++ b/libdeno/deno.gni
@@ -31,7 +31,7 @@ template("create_snapshot") {
                              "testonly",
                              "deps",
                            ])
-    tool = ":snapshot_creator"
+    tool = "//libdeno:snapshot_creator"
     visibility = [ ":*" ]  # Only targets in this file can depend on this.
     snapshot_out_bin = "$target_gen_dir/snapshot_$name.bin"
     inputs = [

--- a/tools/format.py
+++ b/tools/format.py
@@ -26,7 +26,8 @@ qrun([clang_format_path, "-i", "-style", "Google"] +
      find_exts(["libdeno"], [".cc", ".h"]))
 
 print "gn format"
-for fn in ["BUILD.gn", ".gn"] + find_exts(["build_extra"], [".gn", ".gni"]):
+for fn in ["BUILD.gn", ".gn"] + find_exts(["build_extra", "libdeno"],
+                                          [".gn", ".gni"]):
     qrun(["third_party/depot_tools/gn", "format", fn], env=google_env())
 
 print "yapf"

